### PR TITLE
ignore exec.ErrDot when starting caddy in background

### PR DESCRIPTION
**Pull request by WeidiDeng**
Fix [6508](https://github.com/caddyserver/caddy/issues/6508).

### CODE REVIEW

File: cmd/commandfuncs.go [hunk 1]
```diff
@@ -74,6 +74,10 @@ func cmdStart(fl Flags) (int, error) {
 	// sure by giving it some random bytes and having it echo
 	// them back to us)
 	cmd := exec.Command(os.Args[0], "run", "--pingback", ln.Addr().String())
```
Comment (mohammed90) [inline cmd/commandfuncs.go:76]: I don't have access to Windows machine with the same issue, but I wondered if using `os.Executable` would fix it. The error message complains about relative path, but `os.Executable` returns the full path. What do you think?

Comment (WeidiDeng) [reply cmd/commandfuncs.go:76]: `os.Executable` is fine. It's used in other places where caddy binary path is needed, i.e. experimental `upgrade` command.

**Comment by mholt** [comment]
But isn't this a security risk? The reason this error is returned is because this may be unexpected. (Sigh, I don't understand why Windows does this. Of course putting binaries in an explicit PATH is a security benefit.)

**Comment by WeidiDeng** [comment]
`os.Arg[0]` is the program name, which may or may not be a relative path. In case it's a relative path, it still points to the path where the executable is. The security risk is only when the executable it points to may not be what we wanted. That has to do with executable resolution in `PWD`.

The only way this path resolve to an executable that' not wanted is very complex and depends too much on timing. Caddy has to start and the original executable get unlinked and a new executable named caddy has to be created before the original caddy gets to the place where it spawns a new caddy process. This is very unlikely to happen in the real world.

I think `caddy start` is a convenient way to run a caddy. Denying user usage simply because it's not put in the `PATH` directory is against simple to use philosophy.

**Comment by mholt** [comment]
Ah, that's a good point, it's always the 0th CLI arg, which I guess makes it more trustworthy.

**Review by mholt — APPROVED** [review]

File: cmd/commandfuncs.go (hunk 1 above)
Comment (mohammed90) [reply cmd/commandfuncs.go:76]: So is this error ignorable without impacting the `Start` later?

Comment (WeidiDeng) [reply cmd/commandfuncs.go:76]: It's in the [official examples](https://pkg.go.dev/os/exec):

```
cmd := exec.Command("prog")
if errors.Is(cmd.Err, exec.ErrDot) {
	cmd.Err = nil
}
if err := cmd.Run(); err != nil {
	log.Fatal(err)
}
```

**Pull request merged**